sw/device/tests/sim_dv/csrng_fuse_en_sw_app_read.c

```diff
@@ -0,0 +1,225 @@
+  CHECK(got.reseed_counter == expected->reseed_counter, "exp: %x, got: %x",
+        expected->reseed_counter, got.reseed_counter);
+  CHECK(got.fips_compliance == expected->fips_compliance, "exp: %x, got: %x",
+        expected->fips_compliance, got.fips_compliance);
+
+  CHECK_ARRAYS_EQ(got.v, expected->v, ARRAYSIZE(expected->v),
+                  "CSRNG internal V buffer mismatch.");
+
+  CHECK_ARRAYS_EQ(got.key, expected->key, ARRAYSIZE(expected->key),
+                  "CSRNG internal K buffer mismatch.");
+}
+
+/**
+ * CTR DRBG Known-Answer-Test (KAT) for INSTANTIATE command.
+ */
+static void fips_instantiate_kat(const dif_csrng_t *csrng, bool fail_expected) {
+  LOG_INFO("%s", __func__);
+
+  CHECK_DIF_OK(dif_csrng_uninstantiate(csrng));
+  const dif_csrng_seed_material_t kEntropyInput = {
+      .seed_material = {0x73bec010, 0x9262474c, 0x16a30f76, 0x531b51de,
+                        0x2ee494e5, 0xdfec9db3, 0xcb7a879d, 0x5600419c,
+                        0xca79b0b0, 0xdda33b5c, 0xa468649e, 0xdf5d73fa},
+      .seed_material_len = 12,
+  };
+  csrng_testutils_cmd_ready_wait(csrng);
+
+  CHECK_DIF_OK(dif_csrng_instantiate(csrng, kDifCsrngEntropySrcToggleDisable,
+                                     &kEntropyInput));
+
+  const dif_csrng_internal_state_t kExpectedState = {
+      .reseed_counter = 1,
+      .v = {0x06b8f59e, 0x43c0b2c2, 0x21052502, 0x217b5214},
+      .key = {0x941709fd, 0xd8a25860, 0x861aecf3, 0x98a701a1, 0x0eb2c33b,
+              0x74c08fad, 0x632d5227, 0x8c52f901},
+      .instantiated = true,
+      .fips_compliance = false,
+  };
+  const dif_csrng_internal_state_t kZeroState = {};
+
+  check_internal_state(csrng, fail_expected ? &kZeroState : &kExpectedState);
+}
+
+/**
+ * CTR DRBG Known-Answer-Test (KAT) for GENERATE command.
+ */
+static void test_fuse_enable(const dif_csrng_t *csrng) {
+  fips_instantiate_kat(csrng, /*fail_expected=*/false);
+  LOG_INFO("%s", __func__);
+
+  uint32_t got[kExpectedOutputLen];
+
+  csrng_testutils_cmd_generate_run(csrng, got, kExpectedOutputLen);
+  csrng_testutils_cmd_generate_run(csrng, got, kExpectedOutputLen);
+  const dif_csrng_internal_state_t kExpectedState = {
+      .reseed_counter = 3,
+      .v = {0xe73e3392, 0x7d2e92b1, 0x1a0bac9d, 0x53c78ac6},
+
+      .key = {0x66d1b85a, 0xc19d4dfd, 0x053b73e3, 0xe9dc0f90, 0x3f015bc8,
+              0x4436e5fd, 0x1cccc697, 0x1a1c6e5f},
+      .instantiated = true,
+      .fips_compliance = false,
+  };
+  check_internal_state(csrng, &kExpectedState);
+
+  // TODO(#13342): csrng does not provide a linear output order. For example,
+  // note the test vector output word order: 12,13,14,15 8,9,10,11 4,5,6,7
+  // 0,1,2,3.
+  const uint32_t kExpectedOutput[kExpectedOutputLen] = {
+      0xe48bb8cb, 0x1012c84c, 0x5af8a7f1, 0xd1c07cd9, 0xdf82ab22, 0x771c619b,
+      0xd40fccb1, 0x87189e99, 0x510494b3, 0x64f7ac0c, 0x2581f391, 0x80b1dc2f,
+      0x793e01c5, 0x87b107ae, 0xdb17514c, 0xa43c41b7,
+  };
```

Comment:
As above, where do these values come from, and can we add a reference to where to find them or how to generate them?